Parse/src/main/java/com/parse/ParseUser.java

```diff
@@ -504,9 +504,12 @@ private void restoreAnonymity(Map<String, String> anonymousData) {
   @Override
   /* package */ void setState(ParseObject.State newState) {
     // Avoid clearing sessionToken when updating the current user's State via ParseQuery result
-    if (isCurrentUser() && getSessionToken() != null
-            && newState.get("sessionToken") == null) {
-      newState = newState.newBuilder().put("sessionToken", getSessionToken()).build();
+    if (isCurrentUser() && getSessionToken() != null && newState.get(KEY_SESSION_TOKEN) == null) {
+      newState = newState.newBuilder().put(KEY_SESSION_TOKEN, getSessionToken()).build();
+    }
+    // Avoid clearing authData when updating the current user's State via ParseQuery result
+    if (isCurrentUser() && getAuthData() != null && newState.get(KEY_AUTH_DATA) == null) {
+      newState = newState.newBuilder().put(KEY_AUTH_DATA, getAuthData()).build();
```

Comment:
Nit: Let's move `isCurrentUser()` check out a level so we don't have to do it twice